SQLiteException on tables upgrade

I've got this exception while upgrading tables (new value in dbVersion). It seems the table name with "-" is not allowed.

`Caused by: android.database.sqlite.SQLiteException: near "-": syntax error (code 1): , while compiling: DROP TABLE assets$en-US
            at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
            at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:887)
            at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:498)
            at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
            at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
            at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:31)
            at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1674)
            at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1605)
            at com.contentful.vault.SqliteHelper.deleteTables(SqliteHelper.java:128)
            at com.contentful.vault.SqliteHelper.onUpgrade(SqliteHelper.java:70)`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,7 +3,7 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Version [0.9.11][unreleased] - (in development)
-- TBA.
+- Fixed: DB upgrade failures.
 
 ## Version [0.9.10] - 2015-09-03
 - NOTE: This release introduces backwards incompatible changes to any existing database schemas, when upgrading make sure to bump the `dbVersion` attribute on your `Space`.

--- a/core/src/main/java/com/contentful/vault/SqliteHelper.java
+++ b/core/src/main/java/com/contentful/vault/SqliteHelper.java
@@ -113,7 +113,7 @@ final class SqliteHelper extends SQLiteOpenHelper {
     List<String> tables = null;
     try {
       if (cursor.moveToFirst()) {
-        tables = new ArrayList<String>();
+        tables = new ArrayList<>();
         do {
           tables.add(cursor.getString(0));
         } while (cursor.moveToNext());
@@ -125,7 +125,7 @@ final class SqliteHelper extends SQLiteOpenHelper {
       db.beginTransaction();
       try {
         for (String table : tables) {
-          db.execSQL("DROP TABLE " + table);
+          db.execSQL("DROP TABLE " + escape(table));
         }
         db.setTransactionSuccessful();
       } finally {

--- a/tests-integration/src/test/java/com/contentful/vaultintegration/UpgradeTest.java
+++ b/tests-integration/src/test/java/com/contentful/vaultintegration/UpgradeTest.java
@@ -1,0 +1,57 @@
+/*
+ * Copyright (C) 2015 Contentful GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.contentful.vaultintegration;
+
+import com.contentful.vault.Space;
+import com.contentful.vault.Vault;
+import com.contentful.vaultintegration.lib.demo.Cat;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = "src/main/AndroidManifest.xml")
+public class UpgradeTest extends BaseTest {
+  @Space(
+      value = "cfexampleapi",
+      models = Cat.class,
+      locales = "en-US"
+  )
+  static class Sp1 { }
+
+  @Space(
+      value = "cfexampleapi",
+      models = Cat.class,
+      locales = "en-US",
+      dbVersion = 2
+  )
+  static class Sp2 { }
+
+  @Override protected void setupVault() {
+    vault = Vault.with(RuntimeEnvironment.application, Sp1.class);
+  }
+
+  @Test public void testUpgrade() throws Exception {
+    Vault v = vault;
+    v.getReadableDatabase();
+    v.releaseAll();
+    v = Vault.with(RuntimeEnvironment.application, Sp2.class);
+    v.getReadableDatabase();
+  }
+}

